pints/_diagnostics.py

```diff
@@ -100,15 +100,56 @@ def reorder_all_params(chains):
     return samples_all
 
 
-def rhat(samples):
+def rhat(chains):
     r"""
-    Calculates :math:`\hat{R} = \sqrt{(W(n - 1) / n + (1 / n) B) / W}` as per
-    [1]_ for a single parameter. It does this after splitting each chain into
-    two.
-    """
-    W = within(samples)
-    B = between(samples)
-    t = len(samples[0])
+    Returns the convergence measure :math:`\hat{R}` according to [1]_.
+
+    :math:`\hat{R}` diagnoses convergence by checking mixing and stationarity
+    of :math:`M` chains (at least two, :math:`M\geq 2`). To diminish the
+    influence of starting values, the first half of each chain is considered
+    as warm up and does not enter the calculation. Subsequently, the truncated
```

Comment:
looking at the code, it looks like it doesn't discard warm-up. Did I miss something?